examples/io_write.rs

```diff
@@ -0,0 +1,40 @@
+use prometheus_client::{encoding::text::encode, metrics::counter::Counter, registry::Registry};
+use std::io::Write;
+
+/// Example showing how one could write to a file or socket instead of a string.
+/// For large metrics registries this will be more memory efficient
```

Comment:
I would assume a module comment would be more appropriate. Feel free to ignore.

```suggestion
//! Example showing how one could write to a file or socket instead of a string.
//! For large metrics registries this will be more memory efficient.
```